lib/req/finch.ex

```diff
@@ -131,12 +131,16 @@ defmodule Req.Finch do
   end
 
   defp finch_stream_into_collectable(req, finch_req, finch_name, finch_options, collectable) do
-    {acc, collector} = Collectable.into(collectable)
     resp = Req.Response.new()
 
     fun = fn
-      {:status, status}, {acc, req, resp} ->
-        {acc, req, %{resp | status: status}}
+      {:status, 200}, {nil, req, resp} ->
+        {acc, collector} = Collectable.into(collectable)
```

Comment:
Technically we should call `collector.(:halt)` in case of an error. It wasn't the case in the existing code, but I assume that's unexpected.

The issue is that `Finch.stream` doesn't return the accumulator in case of errors, so we have no access to `{acc, collector}`. In the previous code this could've been fixed, because we call `Collectable.into(collectable)` at the top, but with the new code it's not possible. However, I think we should only call `Collectable.into(collectable)` if we know the status is 200, since calling `Collectable.into(collectable)` may have side effects (such as creating a file).

@wojtekmach @josevalim any ideas?

Reply:
Maybe we should change Finch to return the accumulator? Or do you mean the Collector never returns the accumulator?

Reply:
> Maybe we should change Finch to return the accumulator?

Yeah, I think that's exactly what we would need, I'm just not sure if that breaking change is acceptable :D

Reply:
See https://github.com/sneako/finch/pull/295.